Compilation error with old CMake due to broken endianness detection

Just found an issue on CMake 3.15.5:
```
[ 61%] Building C object CMakeFiles/event_static.dir/sha1.c.o
[ 62%] Building C object CMakeFiles/event_static.dir/evrpc.c.o
<command-line>: error: macro names must be identifiers
/home/paul/src/firmware/output/build/libevent-openipc-e8313084f9e8b064433cb10eb9a79bf87407fab6/sha1.c:39:2: error: #error "Endianness not defined!"
 #error "Endianness not defined!"
  ^~~~~
```

It turns out we're using [CMAKE_LANG_BYTE_ORDER](https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_BYTE_ORDER.html) to [define endianness for sha1.c](https://github.com/libevent/libevent/blob/master/sha1.c#L32) that was introduced only in CMake 3.20 (libevent's minimal version is `3.1.2`)

What I have to do to fix it properly?
1. Bump minimal version in CMakeLists.txt
2. Use some kind of handwritten code to determine endianness

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -972,6 +972,15 @@ set(SRC_EXTRA
     sha1.c
     evrpc.c)
 
+if("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" LESS 3.20)
+  include(TestBigEndian)
+  TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
+  if(IS_BIG_ENDIAN)
+    set(CMAKE_C_BYTE_ORDER BIG_ENDIAN)
+  else()
+    set(CMAKE_C_BYTE_ORDER LITTLE_ENDIAN)
+  endif()
+endif()
 set_source_files_properties(sha1.c PROPERTIES COMPILE_FLAGS
     -D${CMAKE_C_BYTE_ORDER}=1)
 add_definitions(-DHAVE_CONFIG_H)

